sympy/integrals/heurisch.py

```diff
@@ -558,8 +558,20 @@ def find_non_syms(expr):
         if solution is None:
             return None
         else:
-            solution = [ (k.as_expr(), v.as_expr()) for k, v in solution.items() ]
-            return candidate.subs(solution).subs(list(zip(coeffs, [S.Zero]*len(coeffs))))
+            sol = {}
+            # remove coeff from k in case Floats were involved
+            # issue 8901
+            for k, v in solution.items():
+                k, v = k.as_expr(), v.as_expr()
+                if not k.is_Symbol:
+                    if k.is_Mul:
+                        c, k = k.as_coeff_Mul()
+                        v /= c
```

Comment:
Before I forget: Here are a couple of lines in meijerint.py:1870 that should be rewritten.
```
        ress = [_meijerint_definite_4(g, x) for g in f.args]
        if all(r is not None for r in ress):
```

When studying 8882 I met an Add with 65 arguments for each one of which `_meijerint_definite_4` returns None. The code diligently fills `ress` with 65 Nones wasting a long time even if one call only takes seconds. I am not quite sure which would be the preferred way to make the code return at the first None.